### PR TITLE
Added Elisp version of the kata

### PR DESCRIPTION
**Pull request by MGadhvi**
# READ ME BEFORE SUBMITTING A PR

Please do not submit a PR with your solution to the Gilded Rose Kata. This repo is intended to be used as a starting point for the kata.

- [x] I acknowledge that this PR is not a solution to the Gilded Rose Kata, but an improvement to the template.
- [x] I acknowledge that I have read [CONTRIBUTING.md](https://github.com/emilybache/GildedRose-Refactoring-Kata/blob/main/CONTRIBUTING.md)

## Please provide your PR description below this line

Added elisp version of the kata. PR contains source code, test file with one failing test and readme file

### CODE REVIEW

File: elisp/gilded-rose-test.el
Comment (codecop) [inline elisp/gilded-rose-test.el:1]: Please make this test similar to other sample tests, e.g. https://github.com/emilybache/GildedRose-Refactoring-Kata/blob/main/clojure/test/gilded/core_test.clj
i.e. check that the name of "foo" is not "fixme".

File: elisp/gilded-rose.el [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  "Return the sell-in value of ITEM."
+  (plist-get item :sell-in))
+
+(defun item-quality (item)
+  "Return the quality of ITEM."
+  (plist-get item :quality))
+
+(defun update-quality (item)
+  "Update the quality of the ITEM according to the Gilded Rose rules."
+  (let ((quality (item-quality item))
```
Comment (codecop) [inline elisp/gilded-rose.el:19]: the lets are already a simplification/improvement of the original code. Is it possible to access the value from the struct each time?

File: elisp/gilded-rose.el
Comment (codecop) [inline elisp/gilded-rose.el:1]: This is not the original code. This is a solution. Please do not send solutions. The translation should be a close to the original code as possible.

**Review by codecop — COMMENTED** [review]
Please see review comments.

File: elisp/gilded-rose.el (hunk 1 above)
Comment (MGadhvi) [reply elisp/gilded-rose.el:19]: Thank you. This was the simplest way I could represent the problem with the elisp I know. I'll have a look at getting the value from the struct each time

File: elisp/gilded-rose.el
Comment (MGadhvi) [reply elisp/gilded-rose.el:1]: I assume this is in relation to your previous point? I'll see if I can make it closer to say, the clojure version or common-lisp version.

**Comment by codecop** [comment]
For future there is no need to create a new pr. New commits appear in the pr automatically. Continues in #574

**Pull request closed without merging**